Orange/widgets/data/owneighbors.py

```diff
@@ -88,24 +82,53 @@ def __init__(self):
 
         self.apply_button = gui.auto_apply(self.controlArea, self, commit=self.apply)
 
-    def _set_label_text(self, name):
-        data = getattr(self, name)
-        label = getattr(self, f"{name}_info_label")
-        if data is None:
-            label.setText(f"No {name} instances")
+    def _set_input_summary(self):
+        n_data, n_refs = 0, 0
+        if self.data is not None and self.reference is not None:
+            n_data = len(self.data)
+            n_refs = len(self.reference)
+        elif self.data is not None:
+            n_data = len(self.data)
+        elif self.reference is not None:
+            n_refs = len(self.reference)
         else:
-            pl = "s" if data else ""
-            label.setText(f"{len(data)} {name} instance{pl} on input.")
+            self.info.set_input_summary(self.info.NoInput)
+            return
+
+        inst = f"{n_data} | {n_refs} "
+        if n_data > 0 and n_refs > 0:
+            details = f"{n_data} data instances on input; " \
+                    f"{n_refs} reference instances on input "
+        elif n_data > 0 and n_refs == 0:
+            details = f"{n_data} data instances on input; " \
+                      f"No reference instances on input "
+        elif n_data == 0 and n_refs > 0:
+            details = f"No data instances on input; " \
+                      f"{n_refs} reference instances on input "
+
+        self.info.set_input_summary(inst, details)
+
+    def _set_output_summary(self):
+        if self.data is None:
+            self.Outputs.data.send(None)
```

Comment:
Outputting the data is already handled in `apply()`. Besides, it should not be sent twice, if not necessary.